File: project_euler/Problem019/Problem019.js
function Problem019() {
    let start = 1900
    let end = 2000
    let days = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
    let length = days.length

    let week = 1
    for (let day of days) {
        week += day
    }
    if (!(start % 4) && (start % 100 || !(start % 400))) {
        ++week
    }
    week %= 7

    let result = 0
    for (let i = start + 1; i <= end; ++i) {
        for (let j = 0; j < length; ++j) {
            if (!week) {
                ++result
            }

            week += days[j]
            if (1 === j && (!(start % 4) && (start % 100 || !(start % 400)))) {
                ++week
            }
            week %= 7
        }
    }
    console.log(result)
}

module.exports = {
    Problem019 : Problem019
};

File: project_euler/Problem019/Problem019.java
package project_euler;

public class Problem019 {
    public void solution() {
        int start = 1900;
        int end = 2000;
        int[] days = new int[]{31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
        int length = days.length;

        int week = 1;
        for (int day : days) {
            week += day;
        }
        if (0 == start % 4 && (0 != start % 100 || 0 == start % 400)) {
            ++week;
        }
        week %= 7;

        int result = 0;
        for (int i = start + 1; i <= end; ++i) {
            for (int j = 0; j < length; ++j) {
                if (0 == week) {
                    ++result;
                }

                week += days[j];
                if (1 == j && (0 == i % 4 && (0 != i % 100 || 0 == i % 400))) {
                    ++week;
                }
                week %= 7;
            }
        }
        System.out.println(result);
    }
}
